java-checks/src/main/java/org/sonar/java/checks/SynchronizedClassUsageCheck.java

```diff
@@ -75,11 +76,25 @@ public void visitNode(Tree tree) {
     tree.accept(new DeprecatedTypeVisitor());
   }
 
+
   @Override
   public void leaveNode(Tree tree) {
     exclusions.pop();
   }
 
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    visited.clear();
+    super.scanFile(context);
+  }
+
+  @Override
+  // This specific override is meant to ensure that the visited set is cleared even for scans that do not go through scanFile (such as the sanity test).
```

Comment:
no need for the comment, it is a common pattern from many rules: any collections storing trees should be cleared at the end or beginning of processing of a file.